packages/ui/src/common/hooks/useSignAndSendTransaction.ts

```diff
@@ -1,91 +1,64 @@
 import { SubmittableExtrinsic } from '@polkadot/api/types'
-import { web3FromAddress } from '@polkadot/extension-dapp'
-import { ISubmittableResult } from '@polkadot/types/types'
-import { useActor } from '@xstate/react'
-import BN from 'bn.js'
-import { useCallback, useEffect } from 'react'
-import { Observable } from 'rxjs'
-import { ActorRef, Sender } from 'xstate'
+import { Hash } from '@polkadot/types/interfaces/types'
+import { useCallback, useEffect, useState } from 'react'
+import { ActorRef } from 'xstate'
 
-import { info } from '../logger'
-import { hasErrorEvent } from '../model/JoystreamNode'
 import { Address } from '../types'
 
-import { useObservable } from './useObservable'
-import { useTransactionStatus } from './useTransactionStatus'
+import { useProcessTransaction } from './useProcessTransaction'
+import { useQueryNodeTransactionStatus } from './useQueryNodeTransactionStatus'
 
-interface UseSignAndSendTransactionParams {
+interface Params {
   transaction: SubmittableExtrinsic<'rxjs'> | undefined
   signer: Address
   service: ActorRef<any>
 }
 
-const observeTransaction = (
-  transaction: Observable<ISubmittableResult>,
-  send: Sender<any>,
-  fee: BN,
-  setPending: (b: boolean) => void
-) => {
-  const statusCallback = (result: ISubmittableResult) => {
-    const { status, events } = result
+// Transactions which emit events handled by QueryNode use useSignAndSendQueryNodeTransaction hook
+// that waits for QueryNode confirmation on PROCESSING stage.
+// Other transactions use simplified hook useSignAndSendTransaction which automatically switch
+// from PROCESSING state to SUCCESS/ERROR.
```

Comment:
I chose which transactions emit events handled by QueryNode (and therefore need to use useSignAndSendQueryNodeTransaction hook) based on:
https://github.com/Joystream/joystream/blob/olympia/query-node/manifest.yml